### PR TITLE
GH-383: Multiple sms recipients

### PR DESCRIPTION
**Pull request by jamesmontemagno**
### Description of Change ###

Describe your changes here. 

### Bugs Fixed ###

- Related to issue #383 

Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR.

### API Changes ###

List all API changes here (or just put None), example:

Added: 
 
- string - string Class.Property { get; set; } 
- void Class.Method();

Changed:

 - object Cell.OldPropertyName => object Cell.NewPropertyName

### Behavioral Changes ###

Describe any non-bug related behavioral changes that may change how users app behaves when upgrading to this version of the codebase.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/View/SMSPage.xaml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/View/SMSPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/SmsViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/ViewModel/SmsViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.android.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.ios.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.shared.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/SmsMessage.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/Xamarin.Essentials/SmsMessage.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C21%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808212121123104-484%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=TGWTanQ%2FB9RLS%2FRz1Dz1a40Ixh7Ck8x0ikaltlOH9v0%3D&st=2018-08-21T21%3A17%3A18Z&se=2018-09-21T21%3A22%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by jamesmontemagno** [comment]
Tested on Android: multiple works in Hangouts, but not in messenger app. Will have to document this.
Test on UWP working great.

Will test on iOS tomorrow when I have my mac.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/View/SMSPage.xaml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/View/SMSPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/SmsViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/ViewModel/SmsViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.android.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.ios.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.shared.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/SmsMessage.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/Xamarin.Essentials/SmsMessage.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C21%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808212146329632-484%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=4JzLe6vIFO34OJCcG88WI7s28t3ZJHozwAqD%2FvkYHdo%3D&st=2018-08-21T21%3A42%3A38Z&se=2018-09-21T21%3A47%3A38Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/View/SMSPage.xaml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/View/SMSPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/SmsViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/ViewModel/SmsViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.android.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.ios.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.shared.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/SmsMessage.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/Xamarin.Essentials/SmsMessage.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C23%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808231439411537-484%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=q52Zbt6hbVsUYb4RlqwiCXtnIM%2B%2F5LDgonuJ%2FFjzbtE%3D&st=2018-08-23T14%3A35%3A43Z&se=2018-09-23T14%3A40%3A43Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by jamesmontemagno** [comment]
This is now good to go @Redth

File: Xamarin.Essentials/Sms/Sms.ios.cs [hunk 1]
```diff
@@ -17,8 +18,8 @@ static Task PlatformComposeAsync(SmsMessage message)
             var messageController = new MFMessageComposeViewController();
             if (!string.IsNullOrWhiteSpace(message?.Body))
                 messageController.Body = message.Body;
-            if (!string.IsNullOrWhiteSpace(message?.Recipient))
-                messageController.Recipients = new[] { message.Recipient };
+
+            messageController.Recipients = message.Recipients.ToArray();
```
[the remark this answers is not in the document]
Comment (jamesmontemagno) [reply Xamarin.Essentials/Sms/Sms.ios.cs]: It would be impossible because I make sure ahead of time.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/View/SMSPage.xaml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/View/SMSPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/SmsViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/ViewModel/SmsViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.android.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.ios.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.shared.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/SmsMessage.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/Xamarin.Essentials/SmsMessage.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C24%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808241527317677-484%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=hco6ZO3hmH7uuLOl4u47XheOVPeSB%2F259EachRhpTeA%3D&st=2018-08-24T15%3A23%3A49Z&se=2018-09-24T15%3A28%3A49Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/View/SMSPage.xaml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/View/SMSPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/SmsViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Samples/Samples/ViewModel/SmsViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.android.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.ios.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.shared.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Sms/Sms.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/multi-sms/Xamarin.Essentials/Sms/Sms.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/SmsMessage.xml](https://github.com/xamarin/Essentials/blob/feature/multi-sms/docs/en/Xamarin.Essentials/SmsMessage.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C24%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808241531371285-484%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=cSj%2BQumWZytWoW5tnIplPTEZtMV%2FeBq4MnL2DZ7Fz3E%3D&st=2018-08-24T15%3A27%3A43Z&se=2018-09-24T15%3A32%3A43Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**